src/lib/logical_query_plan/abstract_lqp_node.hpp

```diff
@@ -66,6 +68,14 @@ class AbstractLQPNode : public std::enable_shared_from_this<AbstractLQPNode> {
  public:
   explicit AbstractLQPNode(LQPNodeType node_type);
 
+  // We do not want people to copy an LQP node as a copy would still have the same children. Instead, they should use
+  // clone().
+  AbstractLQPNode(const AbstractLQPNode&) = delete;
+  AbstractLQPNode& operator=(const AbstractLQPNode&) = delete;
+
+  // Creates a deep copy
+  virtual std::shared_ptr<AbstractLQPNode> clone() const;
```

Comment:
any reason why this is called `clone` and not `deep_copy` ?

Reply:
and for operators this is called `recreate` ? Maybe there's a semantical difference here, but to me this all sounds like the same thing to me

Reply:
`clone` vs `deep_copy`: No reason. For me, they mean the same thing. We can do `deep_copy` as well.

`recreate` is a different thing. It does not simply clone/deep copy the objects but modifies them, replacing the parameter placeholders with actual values.

Reply:
Thanks for the clarification.

I just noticed, too, that in all the comments for the `clone` function you keep using the phrase "deep copy" :D Let's see if the others have any strong feelings about the name

Reply:
changed it.